File: src/main/java/com/my/admin/app/commerce/model/document/Order.java
package com.my.admin.app.commerce.model.document;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.util.List;

@Document(collection = "order")
@Data
public class Order implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    private Long id;

    private List<OrderItem> orderItems;

    private OrderType orderType;
}

File: src/main/kotlin/com/bk/reactive/app/commerce/admin/my/model/document/Order.kt
package com.bk.reactive.app.commerce.admin.my.model.document

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "orders")
data class Order(
    @Id
    val id: String?,
    val orderItems: List<OrderItem>,
    val orderType: OrderType
)